src/amount.cpp

```diff
@@ -19,10 +19,10 @@ CFeeRate::CFeeRate(const CAmount& nFeePaid, size_t nSize)
 
 CAmount CFeeRate::GetFee(size_t nSize) const
 {
-    CAmount nFee = nSatoshisPerK*nSize / 1000;
+    CAmount nFee = nSatoshisPerK * nSize / 1000;
 
-    if (nFee == 0 && nSatoshisPerK > 0)
-        nFee = nSatoshisPerK;
+    if (nFee == 0 && nSize != 0 && nSatoshisPerK != 0)
```

Comment:
I don't understand the `!= 0` here.  Why would you turn a small negative fee rate into a small positive one?
@sdaftuar can comment on the vagaries of negative CFeeRates, they are already fraught with peril, but they can exist at least via `prioritisetransaction`.  I think it would be much safer to keep this as `> 0`.